illumos-utils/src/opte/firewall_rules.rs

```diff
@@ -151,10 +155,19 @@ pub fn opte_firewall_rules(
                             direction,
                             filters: {
                                 let mut filters = Filters::new();
+
+                                // Port assignments are incompatible with non
+                                // TCP/UDP protocols.
+                                if matches!(
+                                    proto,
+                                    ProtoFilter::Tcp | ProtoFilter::Udp
+                                ) {
+                                    filters.set_ports(ports.clone());
+                                }
```

Comment:
I'm surprised we haven't run into this one yet -- if someone were to install a rule on `[TCP, ICMP] x [port 80, port 443]` this would previously be installed as the two rules:

* proto = TCP && (port = 80 || port = 443), [valid]
* proto = ICMP && (port = 80 || port = 443). [invalid]

In OPTE, a port match will always return false unless it is applied to TCP/UDP traffic.